apps/smart-contracts/core/test/ArbitrageBroker.test.ts

```diff
@@ -116,43 +116,83 @@ describe('=> ArbitrageBroker', () => {
+      await arbitrageBroker.connect(governance).setMarketValidity(market.address, true)
+      expect(await arbitrageBroker.isMarketValid(market.address)).to.not.eq(false)
+
+      await arbitrageBroker.connect(governance).setMarketValidity(market.address, false)
+
+      expect(await arbitrageBroker.isMarketValid(market.address)).to.eq(false)
+    })
 
-      await arbitrageBroker.connect(governance).setMarketValidity(user.address, false)
+    it('removes collateral and L/S token approvals if setting to false', async () => {
+      await arbitrageBroker.connect(governance).setMarketValidity(market.address, false)
 
-      expect(await arbitrageBroker.isMarketValid(user.address)).to.eq(false)
+      expect(core.collateral.approve).calledWith(market.address, 0)
+      expect(longToken.approve).calledWith(swapRouter.address, 0)
+      expect(shortToken.approve).calledWith(swapRouter.address, 0)
     })
 
-    it('is idempotent', async () => {
-      expect(await arbitrageBroker.isMarketValid(user.address)).to.not.eq(true)
+    it('is idempotent setting to true', async () => {
+      expect(await arbitrageBroker.isMarketValid(market.address)).to.not.eq(true)
 
-      await arbitrageBroker.connect(governance).setMarketValidity(user.address, true)
+      await arbitrageBroker.connect(governance).setMarketValidity(market.address, true)
 
-      expect(await arbitrageBroker.isMarketValid(user.address)).to.eq(true)
+      expect(await arbitrageBroker.isMarketValid(market.address)).to.eq(true)
+      expect(core.collateral.approve).calledWith(market.address, ethers.constants.MaxUint256)
+      expect(longToken.approve).calledWith(swapRouter.address, ethers.constants.MaxUint256)
+      expect(shortToken.approve).calledWith(swapRouter.address, ethers.constants.MaxUint256)
 
-      await arbitrageBroker.connect(governance).setMarketValidity(user.address, true)
+      await arbitrageBroker.connect(governance).setMarketValidity(market.address, true)
 
-      expect(await arbitrageBroker.isMarketValid(user.address)).to.eq(true)
+      expect(await arbitrageBroker.isMarketValid(market.address)).to.eq(true)
+      expect(core.collateral.approve).calledWith(market.address, ethers.constants.MaxUint256)
+      expect(longToken.approve).calledWith(swapRouter.address, ethers.constants.MaxUint256)
+      expect(shortToken.approve).calledWith(swapRouter.address, ethers.constants.MaxUint256)
+    })
+
+    it('is idempotent setting to false', async () => {
+      expect(await arbitrageBroker.isMarketValid(market.address)).to.not.eq(true)
+
+      await arbitrageBroker.connect(governance).setMarketValidity(market.address, true)
+
+      expect(await arbitrageBroker.isMarketValid(market.address)).to.eq(true)
+      expect(core.collateral.approve).calledWith(market.address, 0)
+      expect(longToken.approve).calledWith(swapRouter.address, 0)
+      expect(shortToken.approve).calledWith(swapRouter.address, 0)
```

Comment:
nit: (similarly for above)
```suggestion
      await arbitrageBroker.connect(governance).setMarketValidity(market.address, true)
      expect(await arbitrageBroker.isMarketValid(market.address)).eq(true)
```